### ISSUE TITLE
Set up Initial Master Server Django App

### ISSUE THREAD

**Issue by ellerychan** [brata-hsdc/brata.masterserver#11]
Stand up the basic skeleton for building the new MS functionality within the Python/Django framework.

Originally this was going to get done as part of #4, but I wanted a separate issue that could be completed in short order.

This is a continuation of **Setup Initial Master Server Django App** brata-hsdc/brata#90.

**Comment by ellerychan** [comment]
@jawaad-ahmad:  So does this Pull Request thing work?  I created a Pull Request related to this issue, but there is no evidence that anyone has seen it.  Are there notifications that go out, but they are being ignored?  I just want a thumbs-up to push some new code, and also wanted to go through the Pull Request cycle once.

**Comment by jawaad-ahmad** [comment]
I see your pull request #12 only after you mentioned it here and I went to look at all the pull requests. I saw no e-mail notification of this yet, but mine aren't turned on too loudly.

I think we need to modify the [Quick Start](../wiki/Quick-Start) page to name who we want to review in the pull request comment using `@mentions`. See proposed. Note that I haven't tried this out, so I don't know if it'll work.

I don't want to review #12 yet because it might mark that I've seen it. Please see if you can edit/add a mention to the pull request and I'll let you know if I got notified.

Do we want to set some sort of general guidance regarding time periods in which pull requests should be reviewed (e.g. 48 hours or 72 hours)?

**Comment by ellerychan** [comment]
I did put @jawaad-ahmad in a comment attached to the Pull Request, but not in the original body msg.  I thought that would be sufficient.  I'll add another one.

Seems like it would be curteous to handle pull requests as soon as you can, but _volunteer effort_, etc., so I wouldn't make a rule about it.

**Comment by jawaad-ahmad** [comment]
It looks like issues and pull requests come from the same pool of IDs. Nice. It turns out I got an e-mail for #12 two days ago with the following text:

> `@jawaad-ahmad`, `@DragonmasterJ`, should look at this and approve. Also Jaron, but what is his @ handle?

Next e-mail I got was from last night:

> Yoohoo `@jawaad-ahmad`

So this corresponds to each comment in which you `@mentioned` me.

I'd recommend leaving a pull request sitting out there for at least two days for folks that are busy just so they can get a chance to look at it provided it isn't needed urgently. This gives volunteers sufficient time to review without being pressured to review immediately.

Agreed since this is volunteer that we don't enforce any maximum, but we do need to address what to do if we put code out there that requested reviewers haven't touched. Do we pull it in without the review and hope for the best?

Also who pulls it in? If I worked the issue, should I be the one to pull it in? Probably not.

If @ellerychan is in charge of all of `piservice`, then should he be the one and only one to pull in all `piservice` issues and make sure they work before giving them his blessing?

**Comment by jawaad-ahmad** [comment]
Let's see if this gets reopened if I post a comment. My choices below are to **Comment** or to **Reopen and comment**. Let me press **Comment**.

Hey you did an awesome job! Keep up the good work!!

**Comment by jawaad-ahmad** [comment]
Ok still closed. Now what happens if I press **Reopen and comment**.

You missed a spot.

**Comment by ellerychan** [comment]
After the merge, it said I could safely delete the branch.  What happens to all the stuff attached to it, like all the comments in the Pull Request?  Does that all stay even if the branch goes away?

**Comment by jawaad-ahmad** [comment]
My understanding is yes. A **branch** in Git is merely a pointer to one
specific commit, so by **deleting a brach**, all you're doing is deleting
that pointer. All commits and history along the way are supposed to be
preserved from what I read.

I haven't tried it yet, but from what I've read is that **deleting a
branch** in Git is non-destructive and is the equivalent of marking a
branch obsolete in other VCSes. All you're supposed to lose is the ability
to make more changes on that branch, which is what you intend to do because
you're done with it and want to close it out.

Before doing so, let's play a little and see what happens if you have an
integration change to make. With the Pull Request closed, what happens if
you commit and push another change on the branch?

**Comment by ellerychan** [comment]
I'll try that, but I wasn't thinking about the code changes.  I was thinking about the discussion thread on GitHub that was attached to the Pull Request.

**Comment by DragonmasterJ** [comment]
This may be from old information, but I read that once we merge, we can no longer open the pull request, if that's what @jawaad-ahmad is talking about above (before deleting branch). 

As far as the discussion thread, I think it is attached to the pull request, which shouldn't go away if the branch is deleted.

**Comment by ellerychan** [comment]
Just a note to say that some of this work is currently being advanced along with the stated work in Issue #13.

**Comment by jawaad-ahmad** [comment]
Anything left to do for this, or can we mark this as resolved?

**Comment by ellerychan** [comment]
Done.  Close it.
